azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1422 @@
+            # and separately a map of service accounts in play. We then iterate through desired/target issuer
+            # to service account pairs to see if they are already present, where if not AND the service account
+            # is in play we federate with best attempt.
+            cred_map = {}
+            cluster_svc_acct_map = {}
+            expected_creds = [(oidc_issuer, SERVICE_ACCOUNT_SECRETSYNC), (oidc_issuer, SERVICE_ACCOUNT_DATAFLOW)]
+            for cred in credentials:
+                svc_acct = cred["properties"]["subject"].split(":")[-1]
+                cred_map[(cred["properties"]["issuer"], svc_acct)] = 1
+                cluster_svc_acct_map[svc_acct] = 1
+
+            for exp_cred in expected_creds:
+                if exp_cred not in cred_map and exp_cred[1] in cluster_svc_acct_map:
+                    subject = f"system:serviceaccount:{self.namespace}:{exp_cred[1]}"
+                    try:
+                        # Federate with best attempt.
+                        msi_client.federated_identity_credentials.create_or_update(
+                            resource_group_name=parsed_uami_id["resource_group"],
+                            resource_name=parsed_uami_id["name"],
+                            federated_identity_credential_resource_name=get_fc_name(
+                                cluster_name=self.cluster_name,
+                                oidc_issuer=oidc_issuer,
+                                subject=subject,
+                            ),
+                            parameters={
+                                "properties": {
+                                    "subject": subject,
+                                    "audiences": ["api://AzureADTokenExchange"],
+                                    "issuer": oidc_issuer,
+                                }
+                            },
+                        )
+                    except HttpResponseError as e:
+                        logger.debug(e)
+
+    def deploy(
+        self,
+        instance_name: Optional[str] = None,
+        use_self_hosted_issuer: Optional[bool] = None,
+    ):
+        if not self.connected_cluster.connected:
+            raise ValidationError(f"Cluster {self.connected_cluster.cluster_name} is not connected to Azure.")
+
+        parameters = {
+            "clusterName": {"value": self.cluster_name},
+        }
+        if instance_name:
+            parameters["instanceName"] = {"value": instance_name}
+        deployment_name = default_bundle_name(self.instance_record["name"])
+
+        DEFAULT_CONSOLE.print()
+
+        deployment_work = []
+        if self.template_mode == TemplateMode.LINKED.value:
+            deployment_work.extend(self.template_content.get_split_content())
+        else:
+            deployment_work.append(self.template_content.content)
+        total_pages = len(deployment_work)
```

Comment:
does this total page referring to number of resource to be deployed? or what does page mean in this context?

Reply:
Yes a page is 800 resources.

Reply:
In this context, a page is the number of deployments.